src/Simulation/Simulators/QuantumExecutor/Borrow.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.QuantumExecutor
+{ 
+    public partial class QuantumExecutorSimulator
+    {
+        public class QuantumExecutorSimBorrow : Intrinsic.Borrow
+        {
+            private readonly QuantumExecutorSimulator sim;
+            public QuantumExecutorSimBorrow(QuantumExecutorSimulator m) : base(m){
+                sim = m;
+            }
+
+            public override Qubit Apply()
+            {
+                IQArray<Qubit> qubits = sim.QubitManager.Borrow(1);
+                sim.QuantumExecutor.OnBorrowQubits(qubits);
+                return qubits[0];
+            }
+
+            public override IQArray<Qubit> Apply( long count )
+            {
+                IQArray<Qubit> qubits = sim.QubitManager.Borrow(1);
```

Comment:
Should this be 'count'?

Reply:
Thanks! This is a bug.

---
In reply to: [349396375](https://github.com/microsoft/qsharp-runtime/pull/82#discussion_r349396375) [](ancestors = 349396375)